### ISSUE TITLE
Campaign Title and Cover photo are not being pulled in on social shares

### ISSUE THREAD

**Issue by lizzydivine** [DoSomethingArchive/legacy-website#5439]
When someone posts "Just Don't Wear Blackface" on Facebook, the name of the campaign and the picture don't appear

![](https://slack-files.com/files-tmb/T024GV2BW-F0C6W1WSH-baeffda165/screenshot_2015-10-09_14.59.40_360.png)

**Comment by mikefantini** [comment]
@sbsmith86 do you have thoughts on this one?

i'm still seeing this not working still. however it's working for me on https://www.dosomething.org/campaigns/social-media-makeover?source=campaigns when I just post both links as my status on FB.

do we need to put _all campaigns_ through the FB linter after we create them?

**Comment by sbsmith86** [comment]
@mikefantini something very weird is going on. i'm not seeing any og meta tags on the page, so there actually is no image set for facebook to crawl. It doesn't look like running it through fb linter helps fix it either. See the screenshot below of what I got back when I ran Just Don't Wear Black Face through the linter.

<img width="938" alt="screen shot 2015-10-14 at 1 37 31 pm" src="https://cloud.githubusercontent.com/assets/1700409/10492257/0f8bef8e-7279-11e5-9c6d-3624e0bea289.png">

**Comment by mikefantini** [comment]
hmmm yeah bummer so meta data on the node https://www.dosomething.org/node/5840/edit:

![image](https://cloud.githubusercontent.com/assets/6330971/10492463/46be212e-727a-11e5-8dc2-847da2224383.png)

compared to an old campaign https://www.dosomething.org/node/2544/edit:

![image](https://cloud.githubusercontent.com/assets/6330971/10492478/5f6f9be4-727a-11e5-9233-0d70d8180695.png)

@sbsmith86 @angaither @sergii-tkachenko probably something with translation?

**Comment by mikefantini** [comment]
hmmm although this is showing up on a new campaign I created as an editor on staging:

https://staging.dosomething.org/node/1735/edit?destination=admin/content

**Comment by sbsmith86** [comment]
Also seeing the issue why the `og:image` tag isn't being set. In [this line](https://github.com/DoSomething/phoenix/blob/5896b2823f770c6aa3a1e9cf46d474ef4ae60564/lib/modules/dosomething/dosomething_metatag/dosomething_metatag.module#L80) Is returning nothing so `$image_nid` is not set and we can create the image from it. Seems to be tied to using `LANGUAGE_NONE` key to find the image nid

**Comment by sbsmith86** [comment]
@mikefantini oh yea sorry, I'm specifically talking about the `og:image` tag. I'm not sure on the other tags. Seem to be find (we are pulling a title, and description correctly). But the image tag missing would cause no image to show on facebook.

**Comment by angaither** [comment]
@mikefantini can we try setting those manually for that node?

**Comment by mikefantini** [comment]
@angaither setting the Meta tag info manually or the `og:image` tag?

I'm less concerned with this one campaign and more just want to make sure that things aren't broken on new campaigns in general.

**Comment by angaither** [comment]
@mikefantini i meant the metatag info. image should be pulled from the campaign cover, and I didn't _think_ those were translatable. but it seems that has changed https://www.dosomething.org/admin/structure/types/manage/campaign/fields/field_image_campaign_cover

**Comment by angaither** [comment]
so [this](https://github.com/DoSomething/phoenix/blob/5896b2823f770c6aa3a1e9cf46d474ef4ae60564/lib/modules/dosomething/dosomething_metatag/dosomething_metatag.module#L80) line will need to be updated to get the language of the node to get the correct image to share

**Comment by angaither** [comment]
@mikefantini are we still seeing this issue?

**Comment by mikefantini** [comment]
@angaither just getting to this.

Yeah this is still an issue. Definitely something we should look at soon. If you don't want it in for 2.1 (which I get) then we should try to fix it early January.

Seeing issues on https://www.dosomething.org/us/campaigns/say-it-stars which was just recently created.

This is what I get when I share that:

![image](https://cloud.githubusercontent.com/assets/6330971/11600137/bdee00f2-9a98-11e5-9172-51e8807f0da5.png)

And this is what I get when I share https://www.dosomething.org/us/campaigns/just-dont-wear-blackface:

![image](https://cloud.githubusercontent.com/assets/6330971/11600160/e260c5f0-9a98-11e5-95f2-78be73946e44.png)

It seems like in general Campaign Title is missing AND it seems as if it's no longer pulling in the cover photo but instead is pulling in random reportback images.

**Comment by angaither** [comment]
I think this is a duplicate of #5096

**Comment by sbsmith86** [comment]
@mikefantini update on this one. This seems to be working for the most part, although it is inconsistent. When I try to share a URL and it doesn't pull the right info, I run it through the [Facebook URL Debugger](https://developers.facebook.com/tools/debug/) and it seems to fix things. 

I found a way for us to do this programmatically and we are discussing in #5936 

I also made some updates in #5937 to fix some errors and warnings I was getting back from Facebook. I'm not sure that will completely solve the issue, but it's better.
